### DISCUSSION TITLE
Declarative TileLayer

### DISCUSSION THREAD

**Discussion by fetishcoder** [visgl/deck.gl#7593] [Q&A] [answered]
Hello there!

Would anybody be able to tell me why this does not work?
```
{
  "initialViewState": {
    "longitude": -1.4157267858730052,
    "latitude": 52.232395363869415,
    "zoom": 6.6,
    "minZoom": 5,
    "maxZoom": 15,
    "pitch": 40.5,
    "bearing": -27.396674584323023
  },
  "views": [
    {
      "@@type": "MapView",
      "controller": true
     }
  ],
  "layers": [
    {
          "@@type": "TileLayer",
          "id": "TileLayer",
          "data": "https://c.tile.openstreetmap.org/{z}/{x}/{y}.png",
          "sublayerProps": {
              "@@type": "BitmapLayer",
              "data": null,
              "image":"@@=data",
              "bounds": ["@@=tile.bbox.west", "@@=tile.bbox.south", "@@=tile.bbox.east", "@@=tile.bbox.north"]
          }
      }
  ]
}
```
https://deck.gl/playground/

I have also tried things like
```
 "bounds": ["@@=west", "@@=south", "@@=east", "@@=north"]
```
but no luck...

PS. the images do seem to come in over the network. they are just not displaying...

Thanks for reading!

**Comment by felixpalmer** [comment, marked as answer]
The problem is that to render the images you need to use `renderSubLayers` as shown in https://deck.gl/docs/api-reference/geo-layers/tile-layer however the JSON format doesn't provide a way to do this via the `@@function` syntax. What you could do is register a `renderSubLayers` function in the configuration (https://deck.gl/docs/api-reference/json/conversion-reference#functions-and-using-function) and then use this. This won't work if you're just using the online playground

**Comment by fetishcoder** [comment]
Much appreciated!
